File: Pulso/department/models/department.py
from django.db import models
from django.core.exceptions import ValidationError
# from accounts.models import Manager
import re

class Department(models.Model):
    name = models.CharField(max_length=100, unique=True)
    active = models.BooleanField(default=True)
    manager = models.ForeignKey("accounts.Manager", on_delete=models.SET_NULL, null=True, blank=True, related_name='departments')

    def clean(self):
        errors = {}

        if not re.match(r'^[A-Za-z0-9\s]+$', self.name):
            errors['name'] = ('O nome deve conter apenas letras, números e espaços.')

        if errors:
            raise ValidationError(errors)
        
    def save(self, *args, **kwargs):
        self.full_clean()
        super().save(*args, **kwargs)

    def __str__(self):
        return self.name

File: src/main/java/dev/pulso/pulso/hospital/model/Department.java
package dev.pulso.pulso.hospital.model;

import dev.pulso.pulso.account.model.Manager;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;

@Entity
@Table(name = "departments")
public class Department {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    @NotBlank(message = "O nome é obrigatório")
    @Pattern(regexp = "^[A-Za-z0-9\\s]+$", message = "O nome deve conter apenas letras, números e espaços.")
    private String name;

    @Column(nullable = false)
    private Boolean active = true;

    @ManyToOne
    @JoinColumn(name = "manager_id")
    private Manager manager;

    public Department() {}

    public Department(String name, Boolean active, Manager manager) {
        this.name = name;
        this.active = active;
        this.manager = manager;
    }

    // Getters e Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public Boolean getActive() { return active; }
    public void setActive(Boolean active) { this.active = active; }

    public Manager getManager() { return manager; }
    public void setManager(Manager manager) { this.manager = manager; }

    @Override
    public String toString() {
        return this.name;
    }
}
